### ISSUE TITLE
[scala, module api] no Callback to get train EvalMetric

### ISSUE THREAD

**Issue by benqua** [apache/incubator-mxnet#5839]
With the scala API, the default output while training with the module api is:
```
2017-04-14 15:16:41,074 [run-main-0] [ml.dmlc.mxnet.module.BaseModule] [INFO] - Epoch[0] Train-mse=0.010826363
2017-04-14 15:16:41,074 [run-main-0] [ml.dmlc.mxnet.module.BaseModule] [INFO] - Epoch[0] Time cost=7032
2017-04-14 15:16:41,773 [run-main-0] [ml.dmlc.mxnet.module.BaseModule] [INFO] - Epoch[0] Validation-mse=0.009271797
```

I would like to plot the training vs validation EvalMetric to see when it starts overfitting.
To do so, I would like to write a cvs vile with the columns epoch, train-mse, validation-mse, because the above output is not really usable to make a plot.

It is possible (with setEvalEndCallbac) to set a BatchEndCallback at the end of the eval stage, to write the validation metric, however, it seems it isn't possible to get the train metric with a callback function. The setEpochEndCallback function takes a EpochEndCallback parameters which doesn't provide a way to access the EvalMetric for the training set.
How can I output the EvalMetric for the training set at the end of an epoch?

My use case (plotting training vs validation error) seems quite common. Is there another (simpler) way to do it? I basically just want the numbers that are output by default, but in a csv format.

**Comment by Ldpe2G** [comment]
@benqua  you can refer to the MnistMlp example: https://github.com/dmlc/mxnet/blob/master/scala-package/examples/src/main/scala/ml/dmlc/mxnetexamples/module/MnistMlp.scala#L43
I think the 'runIntermediateLevelApi' function shows how to get the value of EvalMetric after each epoch.

**Comment by benqua** [comment]
@Ldpe2G I see, but this imply using the intermediate level API.
It would be great to be able to do the same with the high level API (just calling fit on the module) and a callback function, don't you thing so?

**Comment by Ldpe2G** [comment]
Ok, I might understand what you mean. Did you want another EpochEndCallback whose functionality is the same as BatchEndCallback?

@javelinjs what is your opinion?

**Comment by benqua** [comment]
Exactly.

**Comment by javelinjs** [comment]
add `EvalMetric` argument for `EpochEndCallback`?
But the evaluation is done each batch, and the metric will be reset for validation dataset. It may not be clearly defined for what passes to `EpochEndCallback`.

**Comment by benqua** [comment]
@javelinjs not sure to understand your point.
Basically, I just would like to have a way to access the value printed at https://github.com/dmlc/mxnet/blob/master/scala-package/core/src/main/scala/ml/dmlc/mxnet/module/BaseModule.scala#L388 from a callback function.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
